Docs: adds multiple operators example to ppl page (#1040)
pomerium/documentation: content/docs/capabilities/ppl.mdx

title: Policy Language
sidebar_label: Policy Language

> description: Learn how to use Pomerium Policy Language to build context-aware authorization policies for your services.

Author: ```suggestion
description: Learn how to use Pomerium Policy Language to build context-aware authorization policies for routes.
```

  </div>
</details>

> ### Multiple Operators in an Action

Reviewer: Sorry, one more terminology nitpick: do the operators belong to the "action", or to the "rule"?

Based on the above sentence "The object represents a rule where the action is the key and the value is an object containing the logical operators" I wonder if it would be more precise to label this more like "Multiple Operators in One Rule"?

> You _can_ add multiple operators under the same action block, and it would be valid PPL.

Reviewer: (totally optional)

Another phrasing option: "One rule can technically include multiple operators."

For example, this policy would grant access to anyone with a `family_name: Smith` claim, **or** users with email addresses ending in `domain1` or `domain2`:

```yaml
allow:
  and:
    - claim/family_name: Smith
  or:
    - domain:

>         is: domain2.com
>     - domain:
>         is: domain3.com

Reviewer: ```suggestion
        is: domain1.com
    - domain:
        is: domain2.com
```

```

However, you could write an equivalent policy with multiple `allow` blocks:

```yaml
- allow:
    and:
      - claim/family_name: Smith
- allow:
    or:
      - domain:
          is: domain1.com
      - domain:
          is: domain2.com
```

> Although these policies are equally effective, the second example using multiple actions is the intended method to express PPL.

Reviewer: (optional)

Another phrasing suggestion (if you think it would be helpful): "For clarity, we recommend using just one operator per rule."

  </div>
</details>

### Multiple Operators in an Action

> You _can_ add multiple operators under the same Action block, and it would be valid PPL.

Reviewer: nit: "Action" -> "action" (for consistency with the rest of the body text)

> For example, this policy would grant access to anyone with an `example<X>.com` domain in their email address:

Reviewer: I wonder if this `example<X>.com` is a little confusing? It almost reads like any character could substitute for the \<X\>, which is not the case.

It might be clearer to write out "an email address with the domain example1.com, example2.com, or example3.com".

```yaml
allow:
  and:
    - domain:
        is: example1.com
  or:
    - domain:
        is: example2.com
    - domain:
        is: example3.com
```

> However, you could write an identical policy with multiple `allow` blocks:

Reviewer: I think it might be clearer to say "equivalent" rather than "identical".

```yaml

> allow:
>   and:
>     - domain:
>         is: example1.com
> allow:
>   or:
>     - domain:
>         is: example2.com
>     - domain:
>         is: example3.com

Reviewer: To include multiple `allow` actions I think you'd need to make this into a sequence.
```suggestion
- allow:
    and:
      - domain:
          is: example1.com
- allow:
    or:
      - domain:
          is: example2.com
      - domain:
          is: example3.com
```

```

> Although these policies are equally effective, the second example using multiple Actions is the intended method to express PPL.

Reviewer: I'm not sure this makes for a good example. This policy could be further simplified using a single `or` operator, right?

```yaml
allow:
  or:
    - domain:
        is: example1.com
    - domain:
        is: example2.com
    - domain:
        is: example3.com
```
